lib/tasks/reference_data.rake

```diff
@@ -0,0 +1,213 @@
+# frozen_string_literal: true
+
+HESA_CSV_PATH = Rails.root.join("data/hesa/").freeze
+
+HesaCode = Struct.new(:code, :label)
+
+class HesaCodeChecker
+  def initialize
+    @missing = []
+  end
+
+  def call(file_name)
+    puts("## #{file_name.humanize}")
+    puts("Checking HESA codes against Register reference data...")
+
+    puts("| HESA Code | HESA Label | Register Value |")
+    puts("| --------- | ---------- | -------------- |")
+
+    CSV.parse(File.read(HESA_CSV_PATH.join("#{file_name}.csv")), headers: true).each do |row|
+      hesa_code = HesaCode.new(row["Code"], row["Label"])
+      register_value = yield(hesa_code)
+      @missing << hesa_code if register_value.nil?
+      puts("| #{hesa_code.code} | #{hesa_code.label} | #{register_value || 'NOT FOUND'} |")
+    end
+
+    if @missing.any?
+      puts("### Missing mappings 🚨")
+
+      puts("| HESA Code | HESA Label |")
+      puts("| --------- | ---------- |")
+
+      @missing.each do |hesa_code|
+        puts("| #{hesa_code.code} | #{hesa_code.label} |")
+      end
+    else
+      puts("\n")
+      puts("*All HESA codes are mapped!* 🎉")
+    end
+    puts("\n\n")
+  end
+end
+
+namespace :reference_data do
+  desc "All HESA reference data checks"
+  task all: %i[
+    ethnicities
+    sexes
+    countries
+    disabilities
+    itt_aims
+    itt_qualification_aims
+    training_routes
+    course_subjects
+    study_modes
+    age_ranges
+    fund_codes
+    funding_methods
+    training_initiatives
+    degree_types
+    degree_subjects
+    degree_grades
+    degree_institutions
+    degree_countries
+  ]
+
+  desc "HESA ethnicities reference data checks"
+  task ethnicities: :environment do
+    HesaCodeChecker.new.call("ethnicities") do |hesa_code|
+      code_set_key = Hesa::CodeSets::Ethnicities::MAPPING[hesa_code.code]
+      code_set = code_set_key ? CodeSets::Ethnicities::MAPPING[code_set_key] : nil
```

Comment:
Should this return the value of `Hesa::CodeSets::Ethnicities::MAPPING[hesa_code.code]` regardless based on https://github.com/DFE-Digital/register-trainee-teachers/pull/5128 ?